src/ORM/FieldType/DBForeignKey.php

```diff
@@ -96,7 +96,7 @@ public function scaffoldFormField($title = null, $params = null)
             // Remove distinct. Applying distinct shouldn't be required provided relations are not applied.
             $dataQuery->setDistinct(false);
 
-            $dataQuery->setSelect(['over_threshold' => 'count(*) > ' . (int) $threshold]);
+            $dataQuery->setSelect(['over_threshold' => '(CASE WHEN count(*) > ' . (int)$threshold . ' THEN 1 ELSE 0 END)']);
```

Comment:
since we now have a fixed `int` type, then we don't have to check for boolean results anymore below in the line 103?

Reply:
Make sense for me too, do you want to modify my contribution?